tensorflow/java/src/main/java/org/tensorflow/Graph.java

```diff
@@ -77,6 +80,22 @@ public Operation operation(String name) {
   }
 
   /**
+   * Returns all operations that exist in the Graph.
+   */
+  public List<Operation> getOperations() {
```

Comment:
Let's make this return an `Iterator<Operation>` instead of a `List<Operation>`. The C API for iteration predates my involvement with the project, so I'm not entirely sure, but it seems that if there was a reason to prefer iteration there (perhaps memory footprint of when traversing very large graphs), that same reason would apply here. 

I'd imagine something like:

```java
private final class OperationIterator implements Iterator<Operation> {
  OperationIterator(Graph g) {
    this.graph = g;
    advance();
  }

  private void advance() {
    Graph.Reference ref = graph.ref();
    long opHandle = 0;
    try {
       // nextOperation would be a method in graph_jni.cc
       opHandle = Graph.nextOperation(ref.nativeHandle(), pos++);
    } finally {
      ref.close();
    }
    op = opHandle == 0 ? new Operation(this.graph, opHandle) : null;
  }
  public boolean hasNext() { return op != null; }
  public Operation next() { 
    Operation ret = op;
    advance();
    return ret;
  }

  private Graph graph;
  private Operation op = null;
  private int pos = 0;
}

public Iterator<Operation> operations() {
  return new OperationIterator(this);
}
```